joomla/joomla-cms
# Improve hasUpdate checks and make sure we don't offer updates to outdated versions

### Summary of Changes

Improve hasUpdate checks and make sure we don't offer downgrades to outdated versions

### Testing Instructions

- check for updates
- go to the database table #__updates 
- manually change the core update versions collum to a lower version than the current.
- go to the updater page (don't hit refresh)
- see the message to update to an lower version than installed

### Expected result

When we have an old version in the cache joomla says it runs the latest version

### Actual result

When we have an older version in the cache Joomla wants to update to that version.

### Documentation Changes Required

none

### Additional info

The behavior we build by manually changing the database table can happen in case there is some caching of the old version of the update for some reason. one of many examples (german): https://forum.joomla.de/thread/8109-joomla-update-wird-nicht-angeboten-bzw-falsches-update-3-9-10/
